### PR TITLE
Keyword highlighting is unnoticeable in dark theme

### PR DESCRIPTION
**Pull request by balajikris**
This fixes internal bug 1138938 and [Connect bug](https://connect.microsoft.com/VisualStudio/feedback/details/1177735/connect-suggestion-unnoticeable-highlight-color-with-dark-theme)

The issue was that keyword highlighting was using brace highlighting tag
instead of reference highlighting tag (my guess is because brace
highlighting and keyword highlighting were written before reference
highlighting).

This change uses the reference highlighting tag. I also renamed the
keyword lighting folder structure to be called KeywordHighlighting
instead of just Highlighting.

image

### CODE REVIEW

**Comment by balajikris** [comment]
@DustinCampbell @Pilchie @dpoeschl  - Interested in taking a look? 

Cc : @brettfo @basoundr @rchande @jmarolf

File: src/EditorFeatures/Core/EditorFeatures.csproj [hunk 1]
```diff
@@ -390,12 +390,12 @@
...
-    <Compile Include="Implementation\Highlighting\HighlighterViewTaggerProvider.cs" />
-    <Compile Include="Implementation\Highlighting\HighlightingService.cs" />
-    <Compile Include="Implementation\Highlighting\HighlightingTagSource.cs" />
-    <Compile Include="Implementation\Highlighting\HighlightTag.cs" />
+    <Compile Include="Implementation\KeywordHighlighting\AbstractKeywordHighlighter.cs" />
+    <Compile Include="Implementation\KeywordHighlighting\HighlighterTagProducer.cs" />
+    <Compile Include="Implementation\KeywordHighlighting\HighlighterViewTaggerProvider.cs" />
+    <Compile Include="Implementation\KeywordHighlighting\HighlightingService.cs" />
+    <Compile Include="Implementation\KeywordHighlighting\HighlightingTagSource.cs" />
+    <Compile Include="Implementation\KeywordHighlighting\HighlightTag.cs" />
```
Comment (dpoeschl) [inline src/EditorFeatures/Core/EditorFeatures.csproj]: Thank you for renaming the folder! :smile:

**Comment by DustinCampbell** [comment]
:+1:

File: src/EditorFeatures/Core/EditorFeatures.csproj (hunk 1 above)
Comment (balajikris) [inline src/EditorFeatures/Core/EditorFeatures.csproj]: I would have liked to rename all files, namespaces and types appropriately, but the changes are spread all over and have to update all tests. In the end, settled for a cheaper alternative to just rename this folder :wink:

**Comment by dpoeschl** [comment]
:+1: especially because I was going to have to do this as part of 927616 :smile:

**Comment by balajikris** [comment]
Thanks @DustinCampbell, you were spot on about the problem here, in the notes you left in the bug. :smile:

**Comment by basoundr** [comment]
:+1:

**Pull request merged**